5.x - Add docs for assert() usage and overall type improvements. (#7482)
cakephp/docs: en/appendices/5-0-migration-guide.rst

New Features
============

Improved type checking
-----------------------

> CakePHP 5.0 leverages the expanded type system feature available in PHP8.1+.

Reviewer: ```suggestion
CakePHP 5 leverages the expanded type system feature available in PHP8.1+.
```

CakePHP also uses ``assert()`` to provide improved error messages and additional

> type soundness. In production mode you can configure PHP to skip generating

Reviewer: ```suggestion
type soundness. In production mode, you can configure PHP to not generate
```

code for ``assert()`` yielding improved application performance. See the

> :ref:`symlink-assets` for how to do this..

Reviewer: ```suggestion
:ref:`symlink-assets` for how to do this.
```
